### PR TITLE
Patch and get endpoints experiment v2

### PR DESCRIPTION
**Pull request by cosa65**

### CODE REVIEW

File: src/redux/actions/experiments/reorderSamples.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        error: message,
+      },
+    });
+
+    pushNotificationMessage(
+      'error',
+      message,
+    );
+  }
+};
```
Comment (ivababukova) [inline src/redux/actions/experiments/reorderSamples.js:72]: keep in mind that Pol is refactoring the bits around handling errors in actions and also what is returned by fetchAPI. If he merges first, these will need to be changed according to the new error handling implementation, as well as the error handling in `updateExperiment`.

File: src/redux/actions/experiments/updateExperiment.js [hunk 2]
```diff
@@ -47,38 +52,55 @@ const updateExperiment = (
 
       const json = await response.json();
       throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
+    } else if (config.currentApiVersion === api.V2) {
+      if (_.isUndefined(experimentDiff.sampleIds)) {
```
Comment (ivababukova) [inline src/redux/actions/experiments/updateExperiment.js]: why is this check `isUndefined` necessary?

Comment (ivababukova) [reply src/redux/actions/experiments/updateExperiment.js]: I see that if `_.isUndefined` is false, `fetchAPI` won't be called, but `EXPERIMENTS_UPDATED` and `EXPERIMENTS_SAVED` actions will still be dispatched. Can you explain why is that?

Comment (cosa65) [reply src/redux/actions/experiments/updateExperiment.js]: This handling wasn't well written, we discussed with iva and decided to throw an error if a dev tries to set `sampleIds`/`samplesOrder` through this action in v2 because we want this json array to be handled through specific endpoints (for reordering samples for example) or by the api directly as a side effect to whatever happens to the samples

File: src/redux/actions/experiments/reorderSamples.js (hunk 1 above)
Comment (cosa65) [reply src/redux/actions/experiments/reorderSamples.js:72]: I'll see when I try to merge and ask pol if I run into problems

File: src/redux/actions/projects/updateProject.js [hunk 3]
```diff
@@ -22,7 +25,11 @@ const updateProject = (
   const newProject = mergeObjectWithArrays(currentProject, diff);
 
   try {
-    await dispatch(saveProject(projectUuid, newProject));
+    if (config.currentApiVersion === api.V1) {
+      await dispatch(saveProject(projectUuid, newProject));
+    } else if (config.currentApiVersion === api.V2) {
+      // Dont do any fetch, updating the experiment is enough
+    }
```
Comment (ivababukova) [inline src/redux/actions/projects/updateProject.js:32]: this if statement with empty body looks weird. Why did you add it? Is it just to make the code clearer?

**Review by ivababukova — APPROVED** [review]
